tests/unit/usecases/test_network_statistics_api.py

```diff
@@ -0,0 +1,43 @@
+from unittest.mock import MagicMock, patch
+
+from elasticsearch import exceptions
+from pytest import fixture, raises
+
+from common.errors import ApiError
+from usecases.network_statistics_api import NetworkStatisticsApi
+
+
+class TestNetworkStatisticsApi:
+    @fixture
+    def gateway(self):
+        return MagicMock()
+
+    @patch("usecases.network_statistics_api.logger")
+    def test_get_basic_statistics(self, logger, gateway):
+        api = NetworkStatisticsApi(network_statistics_api_gateway=gateway)
+        gateway.get_transaction_statistics.return_value = {"foo": "bar"}
+        result = api.get_basic_statistics()
+        gateway.get_transaction_statistics.assert_called_once()
+        assert result == {"foo": "bar"}
+
+    @patch("usecases.network_statistics_api.logger")
+    def test_get_basic_statistics_gateway_error(self, logger, gateway):
+        api = NetworkStatisticsApi(network_statistics_api_gateway=gateway)
+        gateway.get_transaction_statistics.side_effect = exceptions.ApiError(
+            "gateway_error", MagicMock(), MagicMock()
+        )
```

Comment:
💬 I would suggest here to not use the "gate_error" as message but another text, and use the "gateway_error" to assert the error raised in NetworkStatisticsApi. This one should contain the "gate_error" message. What do you think?

If you agree, it should be applied in the next test as well.

Reply:
Done in https://github.com/HathorNetwork/hathor-explorer-service/pull/283/commits/b0d8d1a1916916621bbca8b20e94494f4d3649c5

I had to change the structure a little bit to be able to assert the message of the raised Exception.